src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2416,4 +2421,20 @@ private EndpointState removeEndpointState(InetAddressAndPort endpoint)
         return removedState;
     }
 
+    private void tryTaskLock()
+    {
+        boolean result;
+        try
+        {
+            result = taskLock.tryLock(LOCK_TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Comment:
timeout is a safety valve - we don't want to pause forever in case of a deadlock